src/World.cpp

```diff
@@ -37,23 +45,12 @@ m_name(j.contains("identifier") ? j["identifier"].get<std::string>() : "")
         nlohmann::json external_level;
         for (const auto& level : j["levels"]) {
             // read then create the external levels
-            std::ifstream in(m_project->getFilePath().directory() + level["externalRelPath"].get<std::string>());
-            if (in.fail()) {
-                ldtk_error("Failed to open file \"" + level["externalRelPath"].get<std::string>() + "\" : " + strerror(errno));
-            }
-            in >> external_level;
+            external_level = nlohmann::json::parse(file_loader(m_project->getFilePath().directory() + level["externalRelPath"].get<std::string>()));
```

Comment:
This line is too long, it is more readable to use an intermediate variable here :

```suggestion
            auto filepath = m_project->getFilePath().directory() + level["externalRelPath"].get<std::string>();
            external_level = nlohmann::json::parse(file_loader(filepath));
```